.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build_lint_test:
+    runs-on: ubuntu-latest
```

Comment:
Nit: I would add a human-readable name to the whole job, rather than having it default to `build_lint_test`:

```suggestion
    name: CI
    runs-on: ubuntu-latest
```